File: app/src/main/java/com/eventhive/localevents/presentation/state/DiscoveryState.kt
package com.eventhive.localevents.presentation.state

enum class DateFilter(val displayName: String) {
    ALL_UPCOMING("All Upcoming"),
    TODAY("Today"),
    TOMORROW("Tomorrow"),
    THIS_WEEK("This Week"),
    THIS_WEEKEND("This Weekend")
}

enum class SortOrder(val displayName: String) {
    SOONEST_FIRST("Soonest First"),
    LATEST_FIRST("Latest First")
}

data class DiscoveryState(
    val searchQuery: String = "",
    val selectedCategory: String = "All",
    val selectedCity: String = "All",
    val selectedNeighborhood: String = "All",
    val selectedDateFilter: DateFilter = DateFilter.ALL_UPCOMING,
    val selectedSortOrder: SortOrder = SortOrder.SOONEST_FIRST
)

File: frontend/src/state/discoveryState.js
export const EVENT_CATEGORIES = Object.freeze([
  'All',
  'Sports',
  'Music',
  'Food',
  'Workshops',
  'Meetups',
  'Student Events',
  'Garage Sale',
  'Community',
])

export const DATE_FILTERS = Object.freeze([
  'All Upcoming',
  'Today',
  'Tomorrow',
  'This Week',
  'This Weekend',
])

export const SORT_ORDERS = Object.freeze(['Soonest First', 'Latest First'])

export const DEFAULT_DISCOVERY_STATE = Object.freeze({
  searchQuery: '',
  selectedCategory: 'All',
  selectedCity: 'All',
  selectedNeighborhood: 'All',
  selectedDateFilter: 'All Upcoming',
  selectedSortOrder: 'Soonest First',
})

export function createDiscoveryState(overrides = {}) {
  return { ...DEFAULT_DISCOVERY_STATE, ...overrides }
}
